platform/darwin/src/MGLStyleLayerManager.h

```diff
@@ -0,0 +1,45 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLStyleLayer.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace mbgl {
+    namespace style {
+        class Layer;
+        class LayerFactory;
+    }
+}
+
+@protocol MGLStyleLayerFactory
+
+/**
+Initializes and returns a layer with a raw pointer to the backing store,
+associated with a style.
+*/
+- (MGLStyleLayer *)createWithRawLayer:(mbgl::style::Layer *)rawLayer;
+
+/**
+ A raw pointer to the mbgl object, which is always initialized.
+ */
+@property (nonatomic, readonly) mbgl::style::LayerFactory *rawLayerFactory;
+
+@end
+
+
+@interface MGLStyleLayerManager : NSObject
+
+/**
+Initializes and returns a layer with a raw pointer to the backing store,
+associated with a style.
+*/
+- (MGLStyleLayer *)createWithRawLayer:(mbgl::style::Layer *)rawLayer;
+
+/**
+ A singleton getter.
+ */
++ (instancetype)get;
```

Comment:
nit: `get` is only used when getting values indirectly.
Rename to `sharedManager` which would bridge to Swift as `shared`.
_Naming conventions in Cocoa/Obj-C are annoyingly verbose._